compiler/typecheck/TcHsType.hs

```diff
@@ -1602,11 +1602,11 @@ tcHsTyVarBndr :: (Name -> Kind -> TcM TyVar)
 --
 -- See also Note [Associated type tyvar names] in Class
 --
-tcHsTyVarBndr new_tv (UserTyVar (L _ name))
+tcHsTyVarBndr new_tv (UserTyVar (L _ name) _)
```

Comment:
This function should really return `TyVarBinder` instead of `TyVar`. That will require a bunch of changes elsewhere to accommodate the new information... like the fact that `tcExplicitTKBndrs` should pass a `[TyVarBinder]` to its inner computation. But I think this will propagate the information to the places that it needs to go... and may reveal bugs if the information ever needs to be ignored.

Reply:
Ah! So *this* might be the missing place (or one of them...).

Reply:
This *is* the missing place.